p2p/protocol/identify/obsaddr.go

```diff
@@ -432,27 +434,89 @@ func (oas *ObservedAddrManager) recordObservationUnlocked(conn network.Conn, obs
 	oas.addrs[localString] = append(oas.addrs[localString], oa)
 }
 
-// 1. If we have an activated address, we are behind an easy NAT.
+// For a given transport Protocol (TCP/UDP):
+//
+// 1. If we have an activated address, we are behind an Cone NAT.
 //
 // 2. If four different peers observe a different address for us on outbound connections, we
-// are MOST probably behind a hard NAT.
+// are MOST probably behind a Symmetric NAT.
 func (oas *ObservedAddrManager) emitNATTypeUnlocked() {
 	var allObserved []*observedAddr
 	for k := range oas.addrs {
 		allObserved = append(allObserved, oas.addrs[k]...)
 	}
 
+	oas.emitTCPNATTypeUnlocked(allObserved)
+	oas.emitUDPTypeUnlocked(allObserved)
+}
+
+func (oas *ObservedAddrManager) emitTCPNATTypeUnlocked(addrs []*observedAddr) {
```

Comment:
this and the subsequent `emitUDPNATTypeUnlocked` are doing basically the same thing. can you factor out a common method that takes an additional parameter (`ma.P_TCP` vs `ma.P_UDP`)?

Reply:
This is done.